coalib/coala.py

```diff
@@ -25,42 +25,35 @@
 from coalib.output.printers.LogPrinter import LogPrinter
 from coalib.parsing.DefaultArgParser import default_arg_parser
 from coalib.settings.ConfigurationGathering import load_configuration
-from coalib.settings.SectionFilling import fill_settings
 
 
 def main():
-    # Note: We parse the args here once to check whether to show bears or not.
-    arg_parser = default_arg_parser()
-    args = arg_parser.parse_args()
-
-    console_printer = ConsolePrinter()
-    if args.show_bears or args.show_all_bears or args.show_language_bears:
+    try:
+        console_printer = ConsolePrinter()
         log_printer = LogPrinter(console_printer)
-        try:
+        # Note: We parse the args here once to check whether to show bears or
+        # not.
+        args = default_arg_parser().parse_args()
+
+        if args.show_bears:
             sections, _ = load_configuration(arg_list=None,
                                              log_printer=log_printer)
-            if args.show_language_bears:
-                local_bears, global_bears = collect_all_bears_from_sections(
-                    sections, log_printer)
+            local_bears, global_bears = collect_all_bears_from_sections(
+                sections, log_printer)
+            if args.filter_by_language is not None:
```

Comment:
`if <variable> is not None` seems more readable.
ᐧ

On Sun, Jun 12, 2016 at 8:09 PM, Lasse Schuirmann <notifications@github.com>
wrote:

> In coalib/coala.py
> <https://github.com/coala-analyzer/coala/pull/2304#discussion_r66721731>:
>
> >              sections, _ = load_configuration(arg_list=None,
> >                                               log_printer=log_printer)
> > -            if args.show_language_bears:
> > -                local_bears, global_bears = collect_all_bears_from_sections(
> > -                    sections, log_printer)
> > +            local_bears, global_bears = collect_all_bears_from_sections(
> > +                sections, log_printer)
> > +            if args.filter_by_language is not None:
>
> hm I guess we don't want to trigger this for empty strings.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/2304/files/670243a1466ec7eb06791e840e1cc2f87ebfdec6..da17ea508375df160563c394ac41afe75c5e336c#r66721731>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOTTomnpe4r0Earsl5DDUgLWpjf1rks5qLBoVgaJpZM4Ivpvk>
> .
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co